src/services/sessionStorage.ts

```diff
@@ -31,38 +24,31 @@ export default class SessionStorage {
         where: { fiscalCode: user.fiscalNumber }
       });
       await loggerUser.createSession({
-        expirationTime: new Date(Date.now() + TOKEN_DURATION_IN_SECONDS),
+        expirationTime: new Date(Date.now() + tokenDurationInSeconds * 1000),
         token: sessionToken
       });
     } catch (error) {
-      return left<Error, boolean>(
-        new Error("Error creating session for the user")
-      );
+      return some<Error>(new Error("Error creating session for the user"));
     }
-    return right<Error, boolean>(true);
+    return none;
   }
 
-  /**
-   * {@inheritDoc}
-   */
   public static async getBySessionToken(
     token: SessionToken
   ): Promise<Either<Error, User>> {
-    const errorOrSession = await this.loadSessionBySessionToken(token);
-
-    if (isLeft(errorOrSession)) {
-      const error = errorOrSession.value;
-      return left(error);
+    try {
+      const user = await User.findOne({
+        include: [{ as: "sessions", model: Session, where: { token } }]
+      });
+      if (user === null) {
+        return left<Error, User>(sessionNotFoundError);
+      }
+      return right<Error, User>(user.get({ plain: true }) as User);
```

Comment:
how do we make sure that the return value of `user.get` can be safely casted to `User` ?